resources/Drivers/SG_Drivers/eeprom/src/eeprom_93aa46.cpp

```diff
@@ -0,0 +1,100 @@
+#include "eeprom_93aa46.hpp"
+
+#include <array>
+
+using namespace sg;
+
+// read can only do one byte at a time on on this device
+Status Eeprom93AA46::read(uint32_t addr, uint8_t* buf, size_t len)
+{
+    sg::Status status;
+
+    for (size_t i = 0; i < len; i++)
+    {
+        status = sendRead(addr + i, buf[i]);
+        if (status != sg::Status::Ok)
+            return status;
```

Comment:
```suggestion
        if (status != sg::Status::Ok)
        {
            return status;
         }
```
Either make a macro or keep check the same as others in the file